### PR TITLE
add some tests

### PR DESCRIPTION
**Pull request by NUicich**

### CODE REVIEW

File: app/api/db/db.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import boto3
+
+class DatabaseOperations:
+    def __init__(self):
+            self.dynamodb = boto3.resource('dynamodb', region_name=os.environ.get("AWS_REGION"))
+            self.football_results = self.dynamodb.Table('bs-football-results')
+            self.football_context_prompts = self.dynamodb.Table('bs-football-context-prompts')
+            self.user_contacts = self.dynamodb.Table('bs-user-contacts')
+
+    @classmethod
```
Comment (TheSlackOne) [inline app/api/db/db.py:16]: Any reason to declare these methods as _static_ (@classmethod)?

File: app/api/db/db.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    def save_prediction(cls, address: str, prediction: str, team: str):
+        """
+        Save prediction to DynamoDB
```
Comment (TheSlackOne) [inline app/api/db/db.py]: No need to address in this PR, but I would use this more _pythonic_ style for documenting:
```
"""
<Brief description>

Parameters:
    @param (param type): <Param description>
    (...)

    Returns:
    <Return description>
"""
```
You can checkout [this code](https://github.com/brainstems/inference-server/blob/main/src/model_operations.py#L9) for reference.

File: app/api/db/db.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        """
+        if os.environ.get('ENVIRONMENT') == 'QA':
+            return { 'prediction': prediction, 'address': address, 'timestamp': timestamp}
+        timestamp = datetime.now(timezone.utc).isoformat()
+        prediction_id = str(uuid.uuid4())
+        try:
+            response = cls().football_results.query(
+                IndexName='address-team-index',
```
Comment (TheSlackOne) [inline app/api/db/db.py:41]: I'm concerned about these hard-coding. This server is thought as a general no-input prediction service, we need to parameterized the fields we read from the DB. We have Trello cards for the Generalization of the services, we can address it later, but it is something that we need to address definitely.

File: app/api/db/db.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                        'id': prediction_id,
+                        'prediction': prediction,
+                        'address': address,
+                        'team': team,
+                        'timestamp': timestamp
+                    }
+                )
+                return {'prediction': prediction, 'address': address, 'timestamp': timestamp}
+            else:
+                return f"Prediction already exists for this address: {address} and team: {team}"
```
Comment (TheSlackOne) [inline app/api/db/db.py:59]: Same comment as above for these. Also for the error string.

File: app/api/predictions/controller.py [hunk 5]
```diff
@@ -43,6 +43,7 @@ async def get_daily_event(
     # api_key: str = Depends(get_api_key)
                           ):
     try:
+
```
Comment (TheSlackOne) [inline app/api/predictions/controller.py]: Probably you don't want this extra line.

File: app/api/predictions/service.py [hunk 6]
```diff
@@ -89,73 +90,44 @@ async def get_new_prediction(cls, prompt: str, client_websocket):
             except Exception as e:
                 print(f"Unable to connect to Inference Server. Retrying in {os.environ.get('RETRY_TIME')} seconds:", e)
                 retry_counts += 1
-                await asyncio.sleep(int(os.environ.get('RETRY_TIME'))) 
+                await asyncio.sleep(int(os.environ.get('RETRY_TIME')))
```
Comment (TheSlackOne) [inline app/api/predictions/service.py]: What if `RETRY_TIME` is not defined? You probably want a fall-back value.

Comment (TheSlackOne) [reply app/api/predictions/service.py]: For line 67 (from a prev PR): 
`ws://{os.environ.get('AKASH_ENDPOINT')}"`
I would parameterized the protocol (`ws`) with a fall-back value.

File: app/api/predictions/service.py [hunk 7]
```diff
@@ -89,73 +90,44 @@ async def get_new_prediction(cls, prompt: str, client_websocket):
...
+        iso_date_str = current_time.isoformat()
+        response = await DatabaseOperations.get_daily_event(iso_date_str)
+        items = response.get('Items', [])
+        if items:
+            # Return the first event from the result
+            return items[0]
+        else:
+            return None
 
     @classmethod
```
Comment (TheSlackOne) [inline app/api/predictions/service.py:108]: Again, we like to understand the reason for this to be static and not others.

File: app/api/predictions/service.py [hunk 8]
```diff
@@ -89,73 +90,44 @@ async def get_new_prediction(cls, prompt: str, client_websocket):
...
+        current_time = datetime.now()
+        iso_date_str = current_time.isoformat()
+        response = await DatabaseOperations.get_next_event(iso_date_str)
+        items = response.get('Items', [])
+        if items:
+            # Find the event with the earliest start_ts
+            next_event = min(items, key=lambda item: item['start_ts'])
+            start_date = next_event['start_ts']
+            team = next_event['team'].replace('_', ' VS ')
+            return {'team': team, 'start_date': start_date}
```
Comment (TheSlackOne) [inline app/api/predictions/service.py:120]: Same for these, we need to param this fields. We can address it later, if so, but we should add it as tech debt. Please, file a card if you won't address this on this PR.

File: app/api/db/db.py (hunk 1 above)
Comment (NUicich) [reply app/api/db/db.py:16]: Static methods have better performance in the entire app. If you don't need an instance of a class, it's better to use static. Instantiating classes consumes memory, which have a lifetime. This way we can avoid that.
(the difference is small but it's better)

File: app/api/db/db.py (hunk 3 above)
Comment (NUicich) [reply app/api/db/db.py:41]: Yes, I agree. The only problem is that each event (football, comic) has its own rules, so it's difficult to generalize it

File: app/api/predictions/controller.py (hunk 5 above)
Comment (NUicich) [reply app/api/predictions/controller.py]: It should be uncommented, but Akash right now doesnt let us send information through the headers. I commented it so the FE can interact with the BE

Comment (TheSlackOne) [reply app/api/predictions/controller.py]: Oh, I mean line 46, the blank one.

**Review by TheSlackOne — APPROVED** [review]

**Pull request merged**